packages/config/src/options/base.js

```diff
@@ -0,0 +1,68 @@
+const { realpath } = require('fs')
+const { dirname, relative } = require('path')
+const { promisify } = require('util')
+
+const pathExists = require('path-exists')
+
+const pRealpath = promisify(realpath)
+
+// Retrieve `base` override.
+// This uses any directory below `repositoryRoot` and above (or equal to)
+// `cwd` that has a `.netlify` or `netlify.toml`. This allows Netlify CLI users
+// to `cd` into monorepo directories to change their base and build directories.
+// Do all checks in parallel for speed
+const getBase = async function({ repositoryRoot, cwd }) {
+  // Performance optimization
+  if (repositoryRoot === cwd) {
+    return
+  }
+
+  const [repositoryRootA, cwdA] = await Promise.all([pRealpath(repositoryRoot), pRealpath(cwd)])
+  const basePaths = getBasePaths(repositoryRootA, cwdA)
+  const basePath = await locatePath(basePaths)
+
+  if (basePath === undefined) {
+    return
+  }
+
+  // `base` starting with a `/` are relative to `repositoryRoot`, so we cannot
+  // return an absolute path
+  return relative(repositoryRoot, dirname(basePath))
+}
+
+// Returns list of files to check for the existence of a `base`
+const getBasePaths = function(repositoryRoot, cwd) {
+  const subdirs = getSubdirs(repositoryRoot, cwd)
+  const basePaths = subdirs.flatMap(subdir => BASE_FILENAMES.map(filename => `${subdir}/${filename}`))
+  return basePaths
+}
+
+// Retrieves list of directories between `repositoryRoot` and `cwd`, including
+// `cwd` but excluding `repositoryRoot`
+const getSubdirs = function(repositoryRoot, dir, subdirs = []) {
+  if (repositoryRoot === dir || repositoryRoot.length > dir.length) {
```

Comment:
Should we check that `dir.startsWith(repositoryRoot)` just in case?

Reply:
Right! Fixed in 29d2c0738a4d58f586445c1c27bf01c29c43304b